### ISSUE TITLE
Strange code in \pandasai\smart_datalake\__init__.py

### ISSUE THREAD

**Issue by PavelAgurov** [gventuri/pandas-ai#467]
### 🐛 Describe the bug

code from class SmartDatalake

```
def chat(self, query: str):
.....

            if result is None:
                self.last_result = result
                self._logger.log(f"Answer: {result}")

```
Logically it should be if `result is not None`. No?

**Comment by gventuri** [comment]
@PavelAgurov definitely a mistake, mind opening a PR for that?

**Comment by PavelAgurov** [comment]
I can if you want. Or just fix it from your side :) It's just one "not". I also trying to understand critical crash - reported in previous bugs.

**Comment by PavelAgurov** [comment]
And many thanks for so fast reaction :)

**Comment by PavelAgurov** [comment]
It seems I have no permissions to create branch.

**Comment by gventuri** [comment]
@PavelAgurov I'm currently not at the computer but I can look into it, if you prefer. It was more a thing that I like to give a chance for everyone to contribute. In order to submit a PR, you need to fork, change on your repo and do A PR, otherwise I can do it if you prefer! Thanks a lot for reporting :)

**Comment by sandiemann** [comment]
@PavelAgurov @gventuri I've added a fix to this along with the property access issue https://github.com/gventuri/pandas-ai/issues/468

**Comment by PavelAgurov** [comment]
Thanks, @sandiemann! 
If you have ideas about this fix https://github.com/gventuri/pandas-ai/issues/465 it will be also very good, because I see at least half of my requests crashed with it.

Next one I will try to fork and contribute by myself ;)

**Comment by gventuri** [comment]
Fixed with #469, closing!
